demisto_sdk/commands/run_unit_tests/unit_tests_runner.py

```diff
@@ -56,25 +52,46 @@ def fix_coverage_report_path(code_directory: Path):
             cursor = sql_connection.cursor()
             files = cursor.execute("SELECT * FROM file").fetchall()
             for id_, file in files:
-                file = Path(file).relative_to("/content")
-                if not (CONTENT_PATH / file).exists():
+                if not file.startswith("/content"):
+                    # means that the .coverage file is already fixed
                     continue
-                cursor.execute(
-                    "UPDATE file SET path = ? WHERE id = ?",
-                    (str(CONTENT_PATH / file), id_),
-                )
+                file = Path(file).relative_to("/content")
+                if (
+                    not (CONTENT_PATH / file).exists()
+                    or file.parent.name
+                    not in file.name  # For example, in `QRadar_v3` directory we only care for `QRadar_v3.py`
+                ):
+                    logger.debug(f"Removing {file} from coverage report")
+                    cursor.execute(
+                        "DELETE FROM file WHERE id = ?", (id_,)
+                    )  # delete the file from the coverage report, as it is not relevant.
+                else:
+                    cursor.execute(
+                        "UPDATE file SET path = ? WHERE id = ?",
+                        (str(CONTENT_PATH / file), id_),
+                    )
             sql_connection.commit()
             logger.debug("Done editing coverage report")
         coverage_file.unlink()
         shutil.copy(temp_file.name, coverage_file)
 
 
 def merge_coverage_report():
-    cov = coverage.Coverage(data_file=CONTENT_PATH / ".coverage")
+    coverage_path = CONTENT_PATH / ".coverage"
+    coverage_path.unlink(missing_ok=True)
+    cov = coverage.Coverage(data_file=coverage_path)
     if not (files := coverage_files()):
         logger.warning("No coverage files found, skipping coverage report.")
         return
-    cov.combine(files, keep=True)
+    fixed_files = []
+    for file in files:
+        try:
+            fix_coverage_report_path(file)
+            fixed_files.append(file)
+        except Exception:
+            logger.warning(f"Broken .coverage file found: {file}, deleting it")
+            Path(file).unlink(missing_ok=True)
```

Comment:
isn't it better to move the try...except logic into fix_coverage_report_path? and return None if not coverage file and leave the code here as list-comp?

Reply:
Greed, done